src/util/misc.rs

```diff
@@ -77,6 +107,55 @@ pub fn script_find_and_remove(haystack: &mut Vec<u8>, needle: &[u8]) -> usize {
     n_deleted
 }
 
+#[cfg(all(test, feature="unstable"))]
+mod benches {
+    use rand::{Rng, thread_rng};
+    use super::hex_bytes;
+    use test::Bencher;
+
+    fn join<I: Iterator<Item=IT>, IT: AsRef<str>>(iter: I, expected_len: usize) -> String {
+        let mut res = String::with_capacity(expected_len);
+        for s in iter {
+            res.push_str(s.as_ref());
+        }
+        res
+    }
+
+    fn bench_from_hex(b: &mut Bencher, data_size: usize) {
+        let data_bytes = thread_rng()
+            .gen_iter()
+            .take(data_size)
+            .collect::<Vec<u8>>();
+        let data = join(data_bytes.iter().map(|x| format!("{:02x}", x)), data_size * 2);
+
+        assert_eq!(hex_bytes(&data).unwrap(), data_bytes);
+
+        b.iter(move || {
```

Comment:
Oops, why move here?

Reply:
Ehh, I'll just merge and fix it in #212.

Reply:
I think that might be a relic from the old bench framework, it complained that the closure lives longer than the reference.